### PR TITLE
Add testing of all PRs on Windows (via Appveyor) WIP

### PR DESCRIPTION
**Pull request by weitzman**
Depends on https://github.com/consolidation/site-process/pull/25

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+install:
+  - cinst -y curl
+  - SET PATH=C:\Program Files\curl;%PATH%
+  #which is only needed by the test suite.
+  - cinst -y which
+  - SET PATH=C:\Program Files\which;%PATH%
+  - git clone -q https://github.com/acquia/DevDesktopCommon.git #For tar, cksum, ...
+  - SET PATH=%APPVEYOR_BUILD_FOLDER%/DevDesktopCommon/bintools-win/msys/bin;%PATH%
+  - SET PATH=C:\Program Files\MySql\MySQL Server 5.7\bin\;%PATH%
+  #Install PHP per https://blog.wyrihaximus.net/2016/11/running-php-unit-tests-on-windows-using-appveyor-and-chocolatey/
```
Comment (ms-slasher13) [inline appveyor.yml]: ERROR: Running ["C:\windows\System32\wusa.exe"  ... wusa must be enabled. 
Should fix the exit code 1058 failures in appveyor build.
`ps: Set-Service wuauserv -StartupType Manual`

**Comment by weitzman** [comment]
I had this PR at green, but am having trouble getting there again after https://github.com/consolidation/site-process/pull/24/ landed. If @ms-slasher13 has any time to work to this, it would be much appreciated.

There are only 2 appveyor fails left:

### testUserCancel

```
1) Unish\UserCase::testUserCancel
Unish\UnishProcessFailedException: Unexpected exit code 1 (expected 0) for command:
C:/projects/work/drush --uri=dev --debug --no-interaction generate content-entity --answers="{""name"":""UnishArticle"",""machine_name"":""unish_article"",""package"":""custom"",""version"":""8.x-1.0-dev"",""dependencies"":""text"",""entity_type_label"":""UnishArticle"",""entity_type_id"":""unish_article"",""entity_base_path"":""admin/content/unish_article"",""fieldable"":""no"",""translatable"":""no"",""revisionable"":""no"",""template"":""no"",""access_controller"":""no"",""title_base_field"":""yes"",""status_base_field"":""yes"",""created_base_field"":""yes"",""changed_base_field"":""yes"",""author_base_field"":""yes"",""description_base_field"":""no"",""rest_configuration"":""no""}" --directory=C:/projects/work/sut/modules/contrib
```

The JSON encoding here is messed up. I think the --directory option is not recognized as separate option due to unbalanced quotes (or something)

### testFailedPostUpdate
```
>  [notice] Update started: woot_update_8104\r\n
>  [notice] This is the update message from woot_update_8104\r\n
>  [notice] Update completed: woot_update_8104\r\n
>  [notice] Update started: woot_post_update_a\r\n
>  [notice] This is the update message from woot_post_update_a\r\n
>  [notice] Update completed: woot_post_update_a\r\n
>  [notice] Update started: woot_post_update_failing\r\n
>  [error]  This is the exception message thrown in woot_post_update_failing \r\n
>  [error]  Update failed: woot_post_update_failing \r\n
\r\n
In ProcessBase.php line 184:\r\n
                                      \r\n
  [InvalidArgumentException]          \r\n
  Unable to decode output into JSON.  \r\n
```

When I comment out [these 2 recently added lines](https://github.com/consolidation/site-process/pull/24/files#diff-d252c6b7b4777df2494d86c94ed603b5R176), it passes. So shouldn't be too hard to fix this one.

File: appveyor.yml (hunk 1 above)
Comment (weitzman) [reply appveyor.yml]: Thanks. Added.

**Comment by ms-slasher13** [comment]
AppVeyor errors:
1) Unish\UserCase::testUserCancel fixed by 
`` 
// Create an isolated input.
            $argv = [
                $generator,
                '--answers=' .  Escape::shellArg($options['answers'], 'LINUX'),
                '--directory=' . $options['directory']
 ];
``
in \Drush\Commands\generate\GenerateCommands on line 71

Seems like a double windows escaping problem.

2) Unish\UpdateDBTest::testFailedPostUpdate fixed by
``
// Try to fix invalid json if needed.
 if (Escape::isWindows() && !json_decode($output, true)) {
``
in \Consolidation\SiteProcess\ProcessBase on line 176

**Many different tools with slightly different json handling.**

**Comment by ms-slasher13** [comment]
see https://github.com/drush-ops/drush/pull/3949

https://ci.appveyor.com/project/weitzman/drush/builds/22712665
One failure left fixed by https://github.com/consolidation/site-process/pull/27

**Comment by weitzman** [comment]
I had to comment out a couple failing tests, but overall I'm happy to see this go in. Our Windows compat should not regress with this in place.

Interested parties are welcome to help with Windows escaping at https://github.com/consolidation/site-process

**Pull request merged**